### DISCUSSION TITLE
Connection reset By peer

### DISCUSSION THREAD

**Discussion by imrrobat** [flet-dev/flet#1713] [General]
hi.. 
i'm trying to see my flet app in phone. 
i installed flet app on phone and run my code in computer with commad:
flet run --android

when i add url of my app in flet app in phone, it says:
Error Connecting to http://..... : Remote end closed connection without response.

can anybody help me to find out problem?

**Comment by FeodorFitsner** [comment]
Can you drop a screenshot with the error?

**Reply by imrrobat** [reply]
hi, yes.. but now Error Change

![photo_2023-08-11_22-19-27](https://github.com/flet-dev/flet/assets/109177330/b42104d2-eaf0-4ba0-8e2b-bf0a2b9f7453)

**Comment by FeodorFitsner** [comment]
Can you open that URL `http://192...` in the browser on your phone?

**Reply by imrrobat** [reply]
yes when i open link on browser on phone, it's automatically open flet app. and then i see this error.

**Reply by FeodorFitsner** [reply]
I mean can you open http://192.168.56.1:8551/my-app/main-py in the browser?

**Reply by DavMelchi** [reply]
Check your laptop IP address and confirm if it is same like the link provided by flet. If not the same, replace the one in the link by your correct IP and try again.

**Reply by imrrobat** [reply]
no.. :( 

![5454](https://github.com/flet-dev/flet/assets/109177330/d595fb52-3612-470c-bfe2-43f810641067)

**Comment by DavMelchi** [comment]
I have got the same issue before and @ndonkoHenri helped me to solve it. It was related to my real local IP address which was not the same like the link provided 
You will find this article useful
https://medium.com/@ndonkohenri/troubleshooting-flet-app-on-mobile-a73b535aa727

Did you check you IP and compare?

**Reply by imrrobat** [reply]
i read this article and yet it didn't work for me... i hot spot my internet from phone to pc. can this cause of problem?

**Reply by DavMelchi** [reply]
Just try now with my phone as hotspot and it works.
I try with another 2nd phone connected to the same hotspot created with the 1st phone and it was a success also.

- link provider by flet qr scan : https://android.flet.dev/192.168.112.1%3A8551/flet_pretest/test_flet-py
- my laptop ip adress : 192.168.67.143
- final link that I used : https://android.flet.dev/192.168.67.143%3A8551/flet_pretest/test_flet-py

Don't forget to change status of your network to private also as indicated in the article 

![image](https://github.com/flet-dev/flet/assets/116588982/713d63d1-2ae3-4fef-83f4-e01a601f77d0)

**Comment by refandas** [comment]
I got the same problem too.
After I checked, it turned out that the IP given by flet was the IP of the WSL, not the IP of the host being used.

![image](https://github.com/flet-dev/flet/assets/37289017/3399a63d-47ca-47d7-8269-ed600d7d1543)
![image](https://github.com/flet-dev/flet/assets/37289017/59280a32-f426-4a44-a624-1ebaebe4aea8)

So, to solve this problem I did:
- I added the project manually by changing the IP to host IP
- Changed the network from "Public" to "Private"
- Because I have an active firewall, so I open firewall on spesific port. For example port 8000, so every time I want to run the flet application, I always use port 8000.

**Reply by imrrobat** [reply]
now i copy my real IP address and put the network on private. and this is a new Error :D 
![photo_2023-08-12_11-25-58](https://github.com/flet-dev/flet/assets/109177330/900f175d-2376-4566-9c59-f11d81c3e742)

**Reply by DavMelchi** [reply]
Copy paste this link in the browser you will have a button to open flet directly. It is the link above the qrcode which does not contain `android.flet.dev` you can add directly manually in flet.

**Reply by imrrobat** [reply]
it still didn't work 😭

**Reply by refandas** [reply]
How if the link is: http://192.168.x.x:8000/my-app/main-py (use http instead of https).

Make sure you have run the app in port 8000 and allow the firewall.
